geode-core/src/main/java/org/apache/geode/internal/cache/versions/RegionVersionVector.java

```diff
@@ -804,26 +811,34 @@ public long getVersionForMember(T mbr) {
    * @return true if this vector has seen the given version
    */
   public boolean contains(T id, long version) {
-    if (id.equals(this.myId)) {
-      if (isForSynchronization()) {
-        // a sync vector only has one holder & no valid version for the vector's owner
+    RegionVersionHolder<T> holder = this.memberToVersion.get(id);
+    // For region synchronization.
+    if (isForSynchronization()) {
+      if (holder == null) {
```

Comment:
The logic should be 

if (isForSynchronization()) { 
} else {
}

Reply:
I think the following code is what we follow to avoid the else block.
if (condition) {
  return sth;
}
....
return sth;